hub/src/server/server.ts

```diff
@@ -59,23 +61,47 @@ export class HubServer {
     const isArchivalRestricted = this.isArchivalRestricted(scopes)
 
     this.validate(address, requestHeaders, oldestValidTokenTimestamp)
-    const listFilesArgs: PerformListFilesArgs = {
-      pathPrefix: address,
-      page: page
-    }
-    if (stat) {
-      const result = await this.driver.listFilesStat(listFilesArgs)
-      if (isArchivalRestricted) {
-        result.entries = result.entries.filter(entry => !this.isHistoricalFile(entry.name))
+    let attemptCount = 0
+    const maxFetchPageAttempts = this.config.fetchPageAttempts || 100
+    
+    const fetchPage = async (pageArg: string): Promise<ListFilesResult | ListFilesStatResult> => {
```

Comment:
I disagree with this approach.  I think that `list-files` should just censor sensitive data, but it shouldn't make the Gaia hub do a bunch of background work like this.  This is just asking for a DDoS.  Instead, it should return a sentinel file name that indicates that the true name is obfuscated or will not be returned.

If the purpose of this code is to make it so blockstack.js's `listFiles()` doesn't return such files to the application, then it should be _blockstack.js_ that re-requests data.

Reply:
Yep -- we want to make sure that we're doing something like a constant number of lookups for any given request, otherwise it's a denial of service vector.

Reply:
I don't really disagree, but clients will end up requesting the hub to do this same thing. I'm okay with blockstack.js being updated to handle the situation. 
@yknl I can do the blockstack.js PR for this -- does that sound good to you?

Reply:
Yea, I think the logical solution here is to have blockstack.js handle empty pages instead of processing it on the hub. 

Just to re-iterate why we shouldn't return `.history` files to blockstack.js `listFile`:
1. It makes `listFiles` on collections almost useless due to the number of irrelevant historical files it returns. 
2. There's a privacy concern with exposing the historical files.

Reply:
PR updated to return a single `null` entry when this situation occurs. blockstack.js needs updated to A) filter these null results from the app and, B) continue to request additional pages.

Reply:
Corresponding blockstack.js PR: https://github.com/blockstack/blockstack.js/pull/685